ElrondNetwork/elrond-go
# add guardian service UID

## Reasoning behind the pull request
Guardian Service UID needs to be linked to the guardian address
  
## Proposed changes
Change the SetGuardian builtin function to have an additional parameter for the ServiceUID and save both the guardian and the service together

## Testing procedure
Use the SetGuardian builtin function to set a guardian both with unguarded (normal) and guarded(co-signed) transactions
Check that the guardian with the aditional service ID has been set in the user account, using the guardian-data api.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
